src/content/docs/guides/lib/packages.md

```diff
@@ -41,4 +41,4 @@ stdenv.mkDerivation {
 ```
 
 This package will be made available on your flake's `packages` output with the same name as the
-directory that you created.
+directory that you created (i.e. `packages.my-package`). Also, from inside a module you can reference it via `pkgs.${namespace}` (i.e. `pkgs.plusultra.my-package`).
```

Comment:
I think we can probably reword this to mention how this is done as a part of overlays which are why system configuration receives these packages as a namespaced collection on `pkgs`.